contracts/Compass.vy

```diff
@@ -232,3 +299,100 @@ def deploy_erc20(_paloma_denom: String[64], _name: String[64], _symbol: String[3
     event_id: uint256 = unsafe_add(self.last_event_id, 1)
     self.last_event_id = event_id
     log ERC20DeployedEvent(_paloma_denom, erc20, _name, _symbol, _decimals, event_id)
+
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # 
+# F E E S
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # 
+
+# Deposit some balance on the contract to be used when sending messages from Paloma.
+# depositor_paloma_address: paloma address to which to attribute the sent amount
+# amount: amount of COIN to register with compass. Overpaid balance will be sent back.
+@external
+@payable
+@nonreentrant('lock')
+def deposit(depositor_paloma_address: bytes32, amount: uint256):
+    if amount > msg.value:
+        send(msg.sender, unsafe_sub(amount, msg.value))
+    else:
+        assert amount == msg.value, "Insufficient deposit"
+    FeeManager(FEE_MANAGER).deposit(depositor_paloma_address, value=amount)
+    log FundsDepositedEvent(depositor_paloma_address, msg.sender, amount)
+
+# Withdraw ramped up claimable rewards from compass. Withdrawals will be swapped and
+# reimbursed in GRAIN.
+# amount: the amount of COIN to withdraw.
+# dex: address of the DEX to use for exchanging the token
+# payload: the function payload to exchange ETH to grain for the dex
+# min_grain: expected grain amount getting from dex to prevent front-running(high slippage / sandwich attack)
+@external
+def withdraw(amount:uint256, dex: address, payload: Bytes[1028], min_grain: uint256):
+    FeeManager(FEE_MANAGER).withdraw(msg.sender, amount, dex, payload, min_grain)
+    log FundsWithdrawnEvent(msg.sender, amount)
+
+# Top up the security funds on the contract used to reimburse all infrastructure
+# related messages. All sent value will be consumed.
+@external
+@payable
+@nonreentrant('lock')
+def security_fee_topup(amount: uint256):
+    if msg.value > amount:
+        send(msg.sender, unsafe_sub(msg.value, amount))
+    else:
+        assert amount == msg.value, "Insufficient deposit"
+    # Make sure we check against overflow here
+    FeeManager(FEE_MANAGER).security_fee_topup(value=amount)
```

Comment:
@webelf101 Since we interface with compass for all these functions that are on the fee manager, would it somehow be possible to get information on user account balances using compass? That would save us a lot of additional work on the protocol.

Reply:
Does this mean view functions? @byte-bandit

Reply:
Yeah, or something that achieves the same result.